### ISSUE TITLE
TypeError when using nested bars in `2.0.0`

### ISSUE THREAD

**Issue by haukebruno** [rsalmei/alive-progress#98]
I tried the `2.0.0` release (or release candidate) today to have nested progress bars.

My testing code is:

```
from time import sleep
from alive_progress import alive_bar

with alive_bar() as customer_bar:
    for customer in ['acmecorp', 'examplecorp', 'foo']:

        print('Start working on customer: ' + customer)

        with alive_bar() as action_bar:
            for action in ['setup', 'install', 'configure', 'finish']:

                print('Performing action ' + action)
                sleep(2)
                action_bar()
        
        sleep(5)
        customer_bar()
```

Which results in an error:

```
File "/Users/hauke/tmp/pybar/foo.py", line 14, in <module>
    print('Performing action ' + action)
  File "/Users/hauke/tmp/pybar/testfoo/lib/python3.9/site-packages/alive_progress/core/hook_manager.py", line 36, in write
    buffer = buffers[stream]
TypeError: unhashable type: 'types.SimpleNamespace'
```

Disclaimer: I'm pretty new to Python (and programming at all) 🙈

**Comment by radugrosu** [comment]
Sorry to comment on an old thread, I understand that nested bars are not supported yet, but it would be great if the application didn't crash.

**Comment by rsalmei** [comment]
No problem, man. Well, if it is not supported it _has_ to crash somehow. You mean, perhaps with a cleaner error message?

**Comment by radugrosu** [comment]
Yes, a more specific error message would actually be great. It happened to me recently that I added a progress bar in a utility where one of the callers was also using a progress bar and it the resulting crash was pretty confusing there for a minute. Thanks

**Comment by rsalmei** [comment]
Done, it will be on the next version: 3.1.3.

<img width="966" alt="image" src="https://github.com/rsalmei/alive-progress/assets/6652853/253ae425-faa9-4df9-a902-012d053e1257">

**Comment by radugrosu** [comment]
Thanks a lot!

**Comment by ulgens** [comment]
Sorry to bring back this issue, but I'm having a case where the crash / error message is not useful.

There are two tasks, let's call them `sync_authors` and `sync_books`, with their own run schedules. Each task has a progress bar. In some cases, `sync_books` encounters unknown author data and calls `sync_authors` to remediate that in the related iteration. That call fails with `Nested use of alive_progress is not yet supported.` and the process fails.

I'm looking for a way to define a progress bar in a way that if it's called nested, it doesn't have to run nested - a safe fallback is enough but we need the code not to crash and iterate the loop. Is that possible with the existing interfaces?

**Comment by ulgens** [comment]
Also, for the paper trail, related commit is https://github.com/rsalmei/alive-progress/commit/76e52ef5f85279e806048505f222729312a220eb
